### PR TITLE
luarocks: Add version 3.8.0

### PR DESCRIPTION
**Pull request by niheaven**
Close https://github.com/ScoopInstaller/Main/issues/1708

### CODE REVIEW

File: bucket/luarocks.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        "   Write-Host 'File' $file 'does not exists. Creating.' -f Yellow",
+        "   $CONT = @(\"rocks_trees = { `\"$rocks`\" }\")",
+        "   Set-Content \"$dir\\$file\" ($CONT -join \"`r`n\") -Encoding Ascii",
+        "}"
+    ],
+    "bin": [
+        "luarocks.exe",
+        "luarocks-admin.exe"
+    ],
+    "env_add_path": "rocks\\bin",
```
Comment (rashil2000) [inline bucket/luarocks.json:31]: I downloaded the archive from https://luarocks.github.io/luarocks/releases/luarocks-3.8.0-windows-64.zip but it does not contain this folder. When is it created?

**Review by rashil2000 — APPROVED** [review]

File: bucket/luarocks.json (hunk 1 above)
Comment (niheaven) [reply bucket/luarocks.json:31]: When you add rocks and if it create bin.

I found this manifest needs more config and test... So convert to draft now

**Comment by rashil2000** [comment]
In this PR can you also fix https://github.com/ScoopInstaller/Main/blob/master/bucket/lua-for-windows.json to change to a more maintained source here https://github.com/wsw0108/scoop-bucket/blob/main/bucket/lua-windows.json

**Comment by niheaven** [comment]
Lua 5.3 is added in https://github.com/ScoopInstaller/Versions/pull/309

**Comment by rashil2000** [comment]
Wait wait, no.

It doesn't make sense for a package in Main bucket (latest, stable versions) to have a dependency in Versions bucket (preview, unstable, older versions).

Besides, having the Lua package is not a hard requirement for Luarocks as people might have installed it through other means.

One more thing - the current package lua-for-windows is very out of date - their last release was in 2018 and it's not properly maintained. You should've added this to the Versions bucket instead, and put lua53 (which is the latest stable Luabinaries release) in the Main bucket.

**Comment by niheaven** [comment]
Lua for Windows is just another Lua distribution that contains many butteries.

The newest version of Lua from Lua Binaries build is 5.4.2 and cannot be used with luarocks or the config file should be modified too much, so I use the past version 5.3 in `Versions`

**Comment by rashil2000** [comment]
Okay, thanks for the info.

So now we can remove `"depends": "versions/lua53",`? 

Or maybe we can change it to `"suggest": "lua",`?

**Comment by niheaven** [comment]
Change to 5.4, and should be merged after https://github.com/ScoopInstaller/Main/pull/2931

`lua` should be dependency, as the install script force luarocks using scooped lua

**Comment by niheaven** [comment]
A simple `luarocks install` result:

image

**Review by rashil2000 — APPROVED** [review]

**Pull request merged**